lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -172,11 +172,14 @@ export class ResponseHandler {
      * @param authority
      */
     addAccountToCache(serverTokenResponse: ServerAuthorizationTokenResponse, idToken: IdToken, authority: Authority): void {
-        const environment = authority.canonicalAuthorityUrlComponents.HostNameAndPort;
-        let accountEntity: AccountEntity;
-        const cachedAccount: AccountEntity = this.uCacheManager.getAccount(this.homeAccountIdentifier, environment, idToken.claims.tid);
+        const accountEntity: AccountEntity = this.generateAccountEntity(
```

Comment:
This change is temporary. Next PR will change `ResponseHandler.ts` to handle silent flows. Please ignore this file in this review.

Reply:
Can you add a TODO here?

Reply:
The PR immediately after addresses this holistically. Since I pull them in sequence, not adding a `TODO`.